object_oriented_programming.ipynb

```diff
@@ -251,25 +310,64 @@
   },
   {
    "cell_type": "markdown",
-   "id": "20",
+   "id": "24",
    "metadata": {},
    "source": [
     "When we ran the above line of code, our object was `r1`, so when `area` was called, Python called the method `area` in the `Rectangle` class automatically, passing `r1` as the argument to the `self` parameter."
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "21",
+   "id": "25",
+   "metadata": {},
+   "source": [
+    "## Example 2\n",
+    "\n",
+    "<div class=\"alert alert-block alert-warning\">\n",
+    "    <h4><b>Question</b></h4>\n",
+    "    Modify your code from <strong>Example 1</strong> so that class <code>Person</code> will now have a <strong>method</strong> called <code>full_name()</code> that returns the string: <code>My name is {first_name} {last_name}</code>. Create an instance of this class, representing a person, which is being initialized by using the arguments passed in the solution function. Lastly, return the instance.\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "26",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%ipytest\n",
+    "\n",
+    "def solution_oop_fullname(first_name: str, last_name: str):\n",
+    "    \"\"\"A fuction that contains the definition of a class Person, and returns an instance of it.\n",
+    "\n",
+    "    Person is a class with two attributes called 'first_name' and 'last_name',\n",
+    "    and a method called 'full_name' which should return the string 'My name is {first_name} {last_name}'.\n",
+    "\n",
+    "    Args:\n",
+    "        first_name: a string used to initialize the Person instance\n",
+    "        last_name: a string used to initialize the Person instance\n",
+    "    Returns:\n",
+    "        - an instance of Person\n",
+    "    \"\"\"\n",
+    "\n",
+    "    # Write your solution here\n",
+    "    pass"
+   ]
+  },
+  {
```

Comment:
I like the new examples with the person, feels intuitive. I added a comment in the test file later for an additional test.
It might also make sense to remind them how importent the self. prefix is when you are using variables of the class itself.